### PR TITLE
support scala 2.13 and scala 3 build for codegen lib

### PR DESCRIPTION
**Pull request by pjfanning**
* upgrades twirl plugin so that we can support scala 3
* CodeGenerator needs to be different in Scala 2.12 (needs a nowarn but this causes newer Scala versions to not allow compilation)
* relates to #180

### CODE REVIEW

File: codegen/src/main/scala-2.13+/org/apache/pekko/grpc/gen/CodeGenerator.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  /** Generator name; example: `pekko-grpc-scala` */
+  def name: String
+
+  def run(request: CodeGeneratorRequest, logger: Logger): CodeGeneratorResponse
+
+  /** Takes Scala binary version and returns suggested dependency Seq */
+  def suggestedDependencies: ScalaBinaryVersion => Seq[Artifact]
+
+  def registerExtensions(registry: ExtensionRegistry): Unit = {}
```
Comment (He-Pin) [inline codegen/src/main/scala-2.13+/org/apache/pekko/grpc/gen/CodeGenerator.scala:35]: Add sclaladoc too？

**Review by He-Pin — APPROVED** [review]
Lgtm

File: codegen/src/main/scala-2.13+/org/apache/pekko/grpc/gen/CodeGenerator.scala (hunk 1 above)
Comment (pjfanning) [reply codegen/src/main/scala-2.13+/org/apache/pekko/grpc/gen/CodeGenerator.scala:35]: the original code (still in use for scala 2.12 compile) does not have scaladoc so I don't want to make the 2 versions differ from each other

Comment (He-Pin) [reply codegen/src/main/scala-2.13+/org/apache/pekko/grpc/gen/CodeGenerator.scala:35]: okay, but you can add doc in both place :)

Comment (pjfanning) [reply codegen/src/main/scala-2.13+/org/apache/pekko/grpc/gen/CodeGenerator.scala:35]: I'm not a fan of hijacking PRs to make unrelated changes - if you want new scaladoc, create a new Issue or PR

**Pull request merged**